app/login.go

```diff
@@ -80,6 +80,7 @@ func (a *App) GetUserForLogin(id, loginId string) (*model.User, *model.AppError)
 
 	// If we are given a userID then fail if we can't find a user with that ID
 	if len(id) != 0 {
+		a.Srv().Store.User().InvalidateProfileCacheForUser(id)
```

Comment:
Do we still need this ? The deep copy takes care of this right ?

Reply:
As far as I understand, yes. I updated the PR message with the reason.